Makefile

```diff
@@ -547,6 +547,41 @@ kind-image: ## Build cilium and operator images and import them into kind.
 	$(MAKE) kind-image-agent
 	$(MAKE) kind-image-operator
 
+.PHONY: kind-install-full
+kind-install-full:
+	@echo "  INSTALL cilium"
+	# cilium-cli doesn't support idempotent installs, so we uninstall and
+	# reinstall here. https://github.com/cilium/cilium-cli/issues/205
+	-cilium uninstall >/dev/null
+	# cilium-cli's --wait flag doesn't work, so we just force it to run
+	# in the background instead and wait for the resources to be available.
+	# https://github.com/cilium/cilium-cli/issues/1070
+	cilium install \
+		--chart-directory=$(ROOT_DIR)/install/kubernetes/cilium \
+		--helm-values=$(ROOT_DIR)/contrib/testing/kind-values.yaml \
+		--version= \
+		>/dev/null 2>/dev/null &
+	for i in seq 1 10; do \
+		if kubectl get -n kube-system ds/cilium 2>/dev/null >/dev/null && \
+		   kubectl get -n kube-system deploy/cilium-operator 2>/dev/null >/dev/null; then \
+		   break; \
+		fi; \
+		sleep 1; \
```

Comment:
Can this 10 second wait loop be nixed for kubectl's `wait` command instead? https://www.mankier.com/1/kubectl-wait

Just removes a heuristic of a ten second wait.

Reply:
Hmm. What I wanted to achieve with this is to make sure that the Cilium resources were installed into the cluster, so that when this returns, `kind-install` target can successfully update the images to point to the debug images.

Does `kind-install` need to set the images to point to the local/debug images? Maybe not. If we rely on this target always installing the local images via `$(ROOT_DIR)/contrib/testing/kind-values.yaml` then the images should be set correctly regardless of how `make kind-install` gets run, so then we can probably skip the wait here and also the image set commands in the `kind-install` target.

Reply:
:bulb: also I just realized that I was assuming `kind-install` is the target to install _cilium_ into kind, but the make target sounds like it's installing kind. I can tweak that too. Maybe `kind-install-cilium` would be less confusing.

Reply:
This will be a nice simplification. I'll test it out:

```
$ git diff HEAD~2
diff --git a/Makefile b/Makefile
index e644a80ce012..6a5b7d5c0d53 100644
--- a/Makefile
+++ b/Makefile
@@ -547,8 +547,8 @@ kind-image: ## Build cilium and operator images and import them into kind.
        $(MAKE) kind-image-agent
        $(MAKE) kind-image-operator
 
-.PHONY: kind-install-full
-kind-install-full:
+.PHONY: kind-install-cilium
+kind-install-cilium: kind-ready ## Install a local Cilium version into the cluster.
        @echo "  INSTALL cilium"
        # cilium-cli doesn't support idempotent installs, so we uninstall and
        # reinstall here. https://github.com/cilium/cilium-cli/issues/205
@@ -561,26 +561,11 @@ kind-install-full:
                --helm-values=$(ROOT_DIR)/contrib/testing/kind-values.yaml \
                --version= \
                >/dev/null 2>/dev/null &
-       for i in seq 1 10; do \
-               if kubectl get -n kube-system ds/cilium 2>/dev/null >/dev/null && \
-                  kubectl get -n kube-system deploy/cilium-operator 2>/dev/null >/dev/null; then \
-                  break; \
-               fi; \
-               sleep 1; \
-       done
-       if ! kubectl get -n kube-system ds/cilium 2>/dev/null >/dev/null || \
-          ! kubectl get -n kube-system deploy/cilium-operator 2>/dev/null >/dev/null; then \
-               echo "failed to install cilium" >&2; \
-               exit 1; \
-       fi;
-
-$(eval $(call KIND_ENV,kind-install))
-kind-install: kind-ready ## Install a local Cilium version into the cluster.
-       $(ECHO_CHECK) cilium is ready...
-       cilium status --wait --wait-duration 1s >/dev/null 2>/dev/null \
-               || $(MAKE) kind-install-full
-       kubectl -n kube-system set image daemonset/cilium cilium-agent=$(LOCAL_AGENT_IMAGE)
-       kubectl -n kube-system set image deployment/cilium-operator cilium-operator=$(LOCAL_OPERATOR_IMAGE)
+
+.PHONY: kind-check-cilium
+kind-check-cilium:
+       @echo "  CHECK  cilium is ready..."
+       cilium status --wait --wait-duration 1s >/dev/null 2>/dev/null
 
 # Template for kind debug targets. Parameters are:
 # $(1) agent target
@@ -604,7 +589,8 @@ kind-debug-agent: ## Create a local kind development environment with cilium-age
        # Not debugging cilium-operator here; any image is good enough.
        $(CONTAINER_ENGINE) push $(LOCAL_OPERATOR_IMAGE) \
                || $(MAKE) kind-image-operator
-       $(MAKE) kind-install
+       $(MAKE) kind-check-cilium \
+               || $(MAKE) kind-install-cilium
        @echo "Attach delve to localhost on these ports to continue:"
        @echo " - 23401: cilium-agent (kind-control-plane)"
        @echo " - 23411: cilium-agent (kind-worker)"
```

Reply:
:heavy_check_mark: seems to do the trick both from fresh install and after running the target before.